docs/design/plugin_multi_fdb_support.md

```diff
@@ -0,0 +1,134 @@
+# Multi FDB cluster support in the kubectl FDB plugin
+
+## Metadata
+
+* Authors: @johscheuer
+* Created: 2021-03-06
+* Updated: 2021-03-06
+
+## Background
+
+The FDB Kubernetes operator supports to setup and operate FDB clusters span across
+multiple Kubernetes clusters. Currently the setup is more or less manual with the
+following steps: bootstrap the seed cluster and once the seed cluster is bootstrapped
+( actually reconciled) we can adjust the database configuration and bootstrap additional
+FDB clusters managed by the Operator. Updating the cluster means adjusting all manifests
+across the Kubernetes clusters at the same time (or in a short time span). Additionally
+the tooling for managing FDB clusters (the kubectl FDB plugin) is currently not aware of
+FDB clusters spread across multiple Kubernetes clusters or even multiple manifests.
+
+## General Design Goals
+
+* Propose a way how to define where a cluster is located.
+* Bootstrap HA FDB clusters with `kubectl fdb`.
+* Make `kubectl fdb` aware of multiple FDB clusters forming an HA cluster.
+
+## Proposed Design
+
+The idea is to use two additional fields in the CRD that will be ignored by the operator but used by the `kubectl fdb plugin`.
+The fields are `seedCluster` and `targetClusters`, both contain the following `cluster objects`:
```

Comment:
We may want to elaborate more on how we plan to support multi-KC clusters where the KC is the primary zone in the replication configuration. I think we would want to cover the following things:

1. Setting the instanceIDPrefix to something separate from the DC ID. I think the best way to do this is to make the name mandatory, and set the instanceIDPrefix to the name, and then have a separate optional field for the name of the cluster object.
2. Setting the zoneIndex and zoneCount fields in the replication configuration. We can probably infer this from the list of targetClusters, but we may also want to allow setting it explicitly to cover the case of moving a part of a cluster from one zone to another.

Reply:
Under the suggestion 1 above, we would also want to validate that the name is unique within the targetClusters list. We could also simplify the seedCluster by having it take just the name instead of the full specification for the cluster.

Reply:
Given how finicky the region configuration structure in FDB is, there may be benefit in adding fields here that can be used to construct the region configuration. This would allow the user to leave the `region` section of the database configuration empty, and only put data center information here.

Reply:
That's a good point I will revisit this part for multi kc